### DISCUSSION TITLE
Yup label not showing

### DISCUSSION THREAD

**Discussion by Maxime-p** [react-hook-form/react-hook-form#9245] [Q&A] [answered]
I'm using Yup as resolver

```js
import { object, string } from "yup";

const validationSchema = object({
    email: string().label("Email address")
        .required("...")
        .email("...")
    password : string().label("Secret password").required("...")
})
```

```js
const { control, handleSubmit, formState: { errors } } = useForm({
    resolver: yupResolver(validationSchema),
    defaultValues: {
        email: '',
        password: ''
    }
});
```

When I console-log errors, I don't see the changes from "label" method
```js
{
  "email": {
    "message": "...",
    "type": "required",
    "ref": {
      "name": "email"
    }
  },
  "password": {
    "message": "...",
    "type": "required",
    "ref": {
      "name": "password"
    }
  }
}
```

![image](https://user-images.githubusercontent.com/43859895/196657829-81a9c9fe-12fb-465e-bea4-5c41f30fcfde.png)

Where I am wrong ?

**Comment by leapful** [comment, marked as answer]
@Maxime-p : It seems you are misunderstanding yup documentation for "Schema.label()". 

The documentation already mentioned label is used to override key name that is being used in error messages, meaning it will be passed when generating error message.

In your code sample, you are using a custom string message with `.require("...")`, so the error message is always fixed as your custom string.

If you want to use label inside that required message, you can use callback instead of string.

For example:

```
yup.string().label('My Label').required(({ label }) => `${label} is required`)
```

After validation, error message will be `My Label is required`.
